link_resolver_track_list.go

```diff
@@ -0,0 +1,210 @@
+			}, nil, noSpecialDur
+		}
+		req.Header.Set("User-Agent", "chatterino-api-cache/1.0 link-resolver")
+
+		// Execute Track list API request
+		resp, err := httpClient.Do(req)
+		if err != nil {
+			return &LinkResolverResponse{
+				Status:  http.StatusInternalServerError,
+				Message: "Track list http request error " + clean(err.Error()),
+			}, nil, noSpecialDur
+		}
+		defer resp.Body.Close()
+
+		// Error out if the track isn't found or something else went wrong with the request
+		if resp.StatusCode < http.StatusOK || resp.StatusCode > http.StatusMultipleChoices {
+			return trackNotFoundResponse, nil, noSpecialDur
+		}
+
+		// Read response into a string
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return &LinkResolverResponse{
+				Status:  http.StatusInternalServerError,
+				Message: "Track list http body read error " + clean(err.Error()),
+			}, nil, noSpecialDur
+		}
+
+		// Parse response into a predefined JSON blob (see TrackListAPIResponse struct above)
+		var jsonResponse TrackListAPIResponse
+		if err := json.Unmarshal(body, &jsonResponse); err != nil {
+			return &LinkResolverResponse{
+				Status:  http.StatusInternalServerError,
+				Message: "Track list api unmarshal error " + clean(err.Error()),
+			}, nil, noSpecialDur
+		}
+		if jsonResponse.Data.ID == 0 { // API responds with {..., "data": null} if nothing was found
+			return trackNotFoundResponse, nil, noSpecialDur
+		}
+
+		trackData := jsonResponse.Data
+
+		prettyAuthors := ""
+		for i, elem := range trackData.Authors {
+			if i != 0 {
+				prettyAuthors += ", "
+			}
+			prettyAuthors += fmt.Sprintf("%s (ID %d - %s)", elem.Name, elem.ID, elem.Role)
+
+		}
+
+		// formatDuration() cannot be use here as that parses an ISO duration
+		duration := time.Duration(trackData.Duration) * time.Second
+		hours := duration / time.Hour
+		duration -= hours * time.Hour
+		minutes := duration / time.Minute
+		duration -= minutes * time.Minute
+		seconds := duration / time.Second
+
+		// Build tooltip data from the API response
+		data := TooltipData{
+			ID:         trackData.ID,
+			Name:       trackData.Name,
+			AuthorName: prettyAuthors,
+			Tags:       strings.Join(trackData.Tags, ", "),
+			Duration:   fmt.Sprintf("%02d:%02d:%02d", hours, minutes, seconds),
+		}
+
+		// Build a tooltip using the tooltip template (see tooltipTemplate) with the data we massaged above
+		var tooltip bytes.Buffer
+		if err := tmpl.Execute(&tooltip, data); err != nil {
+			return &LinkResolverResponse{
+				Status:  http.StatusInternalServerError,
+				Message: "Track list template error " + clean(err.Error()),
+			}, nil, noSpecialDur
+		}
+
+		return &LinkResolverResponse{
+			Status:  200,
+			Tooltip: tooltip.String(),
+			//Thumbnail: thumbnailURL,
+		}, nil, noSpecialDur
+	}
+
+	cache := newLoadingCache("tracklist_tracks", load, 1*time.Hour)
+	trackPathRegex := regexp.MustCompile(`/track/detail/([0-9]+)`)
+
+	// BetterTTV hosts we're doing our smart things on
```

Comment:
Looks like you have one more confusing comment left.

Reply:
changed comment and renamed the variable in `0ed9d81`